cartridges/int_klaviyo_sfra/cartridge/controllers/KlaviyoRecreate.js

```diff
@@ -0,0 +1,96 @@
+ * A DECODED query would be structured as:  <baseURL>/Cart-Recreate?items=[ {"productID": ProdID, "quantity": QTY, "options": [ {lineItemText, optionID, optionValueID} ]} ]
+ * @function
+ * @memberof Cart
+ * @param {querystringparameter} - items - JSON containing product Ids, qty, childProducts, options
+ * @param {renders} - isml
+ * @param {serverfunction} - get
+ */
+server.get('Cart', function (req, res, next) {
+    var currentBasket = BasketMgr.getCurrentOrNewBasket();
+    try {
+        var items = req.querystring.items ? JSON.parse(StringUtils.decodeBase64(req.querystring.items)) : null;
+    } catch (error) {
+        res.setStatusCode(500);
+        res.json({
+            error: true,
+            errorMessage: `ERROR - ${error.message}. Please check the encoded obj for any unexpected chars or syntax issues.`,
+        });
+        return next();
+    }
+
+    if (!currentBasket) {
+        res.setStatusCode(500);
+        res.json({
+            error: true,
+            errorMessage: `ERROR - Current Basket is: ${currentBasket}. Please check the Current Basket and try again.`,
+            redirectUrl: URLUtils.url('Cart-Show').toString()
+        });
+        return next();
+    };
+
+    // Clean the basket to prevent product duplication on page refresh
+    if (currentBasket && currentBasket.productQuantityTotal > 0) {
+        for (let i = 0; i < currentBasket.productLineItems.length; i++) {
+            let lengthAtIteration = currentBasket.productLineItems.length;
+            var shipmentToRemove = currentBasket.productLineItems[i].shipment;
+            currentBasket.removeProductLineItem(currentBasket.productLineItems[i]);
+            currentBasket.updateTotals();
+            if (currentBasket.productLineItems.length < lengthAtIteration) {
+                i--;
+            }
+
+            if (shipmentToRemove.productLineItems.length && !shipmentToRemove.default) {
+                currentBasket.removeShipment(shipmentToRemove);
+            }
+            PromotionMgr.applyDiscounts(currentBasket);
+        }
+    }
+
+    Transaction.wrap(function () {
+        if (items && items.length) {
+            for (let i = 0; i < items.length; i++) {
+                var productToAdd = ProductMgr.getProduct(items[i].productID);
+                var childProducts = productToAdd.bundledProducts ? collections.map(productToAdd.bundledProducts, function (product) { return { pid: product.ID, quantity: null } }) : [];
+                var options = [];
+                items[i].options.forEach(optionObj => {
+                    options.push({ lineItemText: optionObj.lineItemText, optionId: optionObj.optionID, selectedValueId: optionObj.optionValueID});
+                })
+
+                var shipments = Array.from(currentBasket.shipments);
+                shippingHelper.ensureShipmentHasMethod(shipments[0]);
```

Comment:
Is there a reason why ALL shipments in the array shouldn't be checked for a shipping method?

Reply:
No longer creating an array - instead iterating through object directly to remain more dynamic and calling the shippingHelper function on every shipment that already exists. That adjustment is available in an updated PR here:

- https://github.com/maze-consulting/SFCC_Klaviyo_Maze/pull/19